src/turbomind/models/llama/unified_attention_layer.cc

```diff
@@ -144,490 +100,173 @@ inline void UnifiedAttentionLayer<T>::forward(TensorMap* outputs, const TensorMa
 
     /////////////////////////////////////////////
     /// parse inputs
-    const int num_token   = inputs->at("input_query").shape[0];
-    const int layer_id    = inputs->getVal<int>("layer_id");
-    const int session_len = inputs->getVal<int>("session_len");
-
-    int pf_batch_size = 0;
-    int pf_max_q_len  = 0;
-    int pf_max_k_len  = 0;
-    T*  attention_mask{};
-    if (inputs->isExist("attention_mask")) {
-        pf_batch_size  = inputs->at("attention_mask").shape[0];
```

Comment:
Maybe the comments for this function should be updated. The values in the input tensors are changed a lot.